libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -40,18 +41,24 @@ public static EntitlementChecker checker() {
     public static void initialize(Instrumentation inst) throws Exception {
         manager = new ElasticsearchEntitlementChecker();
 
-        // TODO: Configure actual entitlement grants instead of this hardcoded one
-        Method targetMethod = System.class.getMethod("exit", int.class);
-        Method instrumentationMethod = Class.forName("org.elasticsearch.entitlement.bridge.EntitlementChecker")
-            .getMethod("checkSystemExit", Class.class, int.class);
-        Map<MethodKey, Method> methodMap = Map.of(INSTRUMENTER_FACTORY.methodKeyForTarget(targetMethod), instrumentationMethod);
+        Map<MethodKey, CheckerMethod> methodMap = INSTRUMENTER_FACTORY.lookupMethodsToInstrument(
+            "org.elasticsearch.entitlement.bridge.EntitlementChecker"
+        );
 
-        inst.addTransformer(new Transformer(INSTRUMENTER_FACTORY.newInstrumenter("", methodMap), Set.of(internalName(System.class))), true);
-        inst.retransformClasses(System.class);
+        var classesToTransform = methodMap.keySet().stream().map(MethodKey::className).collect(Collectors.toSet());
+
+        inst.addTransformer(new Transformer(INSTRUMENTER_FACTORY.newInstrumenter("", methodMap), classesToTransform), true);
+        // TODO: should we limit this array somehow?
+        var classesToRetransform = classesToTransform.stream().map(EntitlementInitialization::internalNameToClass).toArray(Class[]::new);
+        inst.retransformClasses(classesToRetransform);
     }
 
-    private static String internalName(Class<?> c) {
-        return c.getName().replace('.', '/');
+    private static Class<?> internalNameToClass(String internalName) {
+        try {
+            return Class.forName(internalName.replace('/', '.'), false, ClassLoader.getPlatformClassLoader());
```

Comment:
I think using the platform class loader for classes to instrument is OK?

Reply:
I think so too.